tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs

```diff
@@ -14,45 +14,44 @@ import Plutus.V1.Ledger.Api (
  )
 import Plutus.V1.Ledger.Scripts (
   Context,
-  ScriptError,
+  ScriptError (EvaluationError, EvaluationException, MalformedScript),
   runMintingPolicyScript,
   runScript,
  )
 import Safe (lastMay)
 
--- The result of parsing a log from a script emulation
+-- The result of parsing a log from a errored script emulation
 data ScriptResult
-  = ScriptPassed
-  | ScriptFailed
-  | NoOutcome
-  | ParseFailed Text
-  | InternalError Text
+  = ScriptFailed [Text] String
+  | ParseFailed [Text] Text
+  | PlutusEvaluationException String String
+  | PlutusMalformedScript String
   deriving stock (Eq, Show)
 
 testValidatorScript ::
   Context ->
   Validator ->
   Datum ->
   Redeemer ->
-  Either ScriptError ([Text], ScriptResult)
+  Either ScriptResult [Text]
 testValidatorScript ctx val d r = case runScript ctx val d r of
-  Left err -> Left err
-  Right (_, logs) -> Right . (logs,) . parseLogs $ logs
+  Left err -> Left $ parseError err
+  Right (_, logs) -> Right logs
 
 testMintingPolicyScript ::
   Context ->
   MintingPolicy ->
   Redeemer ->
-  Either ScriptError ([Text], ScriptResult)
+  Either ScriptResult [Text]
 testMintingPolicyScript ctx mp r = case runMintingPolicyScript ctx mp r of
-  Left err -> Left err
-  Right (_, logs) -> Right . (logs,) . parseLogs $ logs
+  Left err -> Left $ parseError err
+  Right (_, logs) -> Right logs
 
-parseLogs :: [Text] -> ScriptResult
-parseLogs logs = case lastMay logs >>= Text.stripPrefix "tasty-plutus: " of
-  Nothing -> NoOutcome
-  Just "Pass" -> ScriptPassed
-  Just "Fail" -> ScriptFailed
-  Just t -> case Text.stripPrefix "Parse failed: " t of
-    Nothing -> InternalError t
-    Just t' -> ParseFailed t'
+parseError :: ScriptError -> ScriptResult
+parseError = \case
+  EvaluationError logs msg ->
+    case lastMay logs >>= Text.stripPrefix "tasty-plutus: Parse failed: " of
+      Nothing -> ScriptFailed logs msg
+      Just t -> ParseFailed logs t
+  EvaluationException name msg -> PlutusEvaluationException name msg
+  MalformedScript msg -> PlutusMalformedScript msg
```

Comment:
The `MalformedScript` constructor is gone from the latest Plutus, you may want to disregard this case.

Reply:
If you do, you'll have a problem finding a compatible version of `plutus-apps`. It's lagging behind `plutus`. I had to use this fork a couple weeks ago: https://github.com/mlabs-haskell/plutus-apps/tree/mario/MalformedScript

Reply:
@blamario From what I could determine, [this commit](https://github.com/input-output-hk/plutus/commit/4127e9cd6e889824d724c30eae55033cb50cbf3e) is what `plutus-apps` is pinned to. It's only 2 weeks old - are the changes you need newer than this, or am I missing something here?

Reply:
No the particular change to `plutus` was older than this. Try your luck with the latest `plutus-apps` then.